lib/benchee/benchmark/runner.ex

```diff
@@ -112,8 +114,7 @@ defmodule Benchee.Benchmark.Runner do
         num_iterations: num_iterations
       }
 
-    updated_scenario = %Scenario{scenario | run_times: [initial_run_time]}
-    do_benchmark(updated_scenario, new_context)
+    do_benchmark(scenario, new_context, [initial_run_time])
```

Comment:
Keeping the `run_times` separate from the `scenario` here is the only thing that feels a little odd to me. I totally get it now, but I also know that like 6 months later when I see this code again I know I'm going to be scratching my head saying "why aren't we putting the run times in the scenario where they belong?"

Maybe a comment in front of `do_benchmark/3` with a little explanation as to why we're doing that would keep me from confusing myself later on?

Reply:
Sounds good to me, I'll write something up!